app/src/test/java/waiter/CommunicatorTest.java

```diff
@@ -1,28 +1,33 @@
 package waiter;
 
-import waiter.ClientConnection.Connectable;
 import waiter.Communicator.Communicator;
 import waiter.Listener.mock.ListenerMock;
 import waiter.Messenger.mock.MessengerMock;
-
-import org.junit.jupiter.api.Test;
 import waiter.Reactor.mock.ReactorMock;
 
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeEach;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 import java.io.IOException;
 
 class CommunicatorTest {
 
+    String clientHasDisconnected;
+
+    @BeforeEach
+    void setUp() {
+        clientHasDisconnected = null;
+    }
+
     @Test
     void communicatesProvidedInput() throws IOException {
-        ListenerMock listenerMock = new ListenerMock("foo");
+        String[] clientRequests = {"curl foo1", "curl foo2", "curl foo3", "curl foo4"};
+        ListenerMock listenerMock = new ListenerMock(new String[]{"foo", clientHasDisconnected});
```

Comment:
What is the significance of this?

Reply:
Hmmm, are you asking this about the strings? Im not sure which part, Im sorry ☹️

Reply:
No worries, I think you answered it in your other comment so it's all good!